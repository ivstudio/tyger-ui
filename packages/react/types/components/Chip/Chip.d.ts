/// <reference types="react" />
export declare type TSize = 'sm' | 'md';
declare type TChipVariant = 'filled' | 'outlined';
declare type TChipTags = 'span' | 'div' | 'button';
interface IChip {
    id?: string;
    tag?: TChipTags;
    size?: TSize;
    label: React.ReactNode;
    onClick?: () => void;
    variant?: TChipVariant;
    className?: string;
    disabled?: boolean;
}
export declare const ChipRoot: import("styled-components").StyledComponent<"div", import("styled-components").DefaultTheme, {
    size: TSize;
    variant: TChipVariant;
    disabled: boolean;
}, never>;
export declare const ChipButtonRoot: import("styled-components").StyledComponent<"div", import("styled-components").DefaultTheme, {
    size: TSize;
    variant: TChipVariant;
    disabled: boolean;
}, never>;
declare const Chip: ({ id, label, size, tag, onClick, variant, className, disabled, }: IChip) => JSX.Element;
export default Chip;
