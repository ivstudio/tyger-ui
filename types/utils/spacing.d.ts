export declare const spacing: {
    readonly '0': "0px";
    readonly '2': "2px";
    readonly '4': "4px";
    readonly '8': "8px";
    readonly '12': "12px";
    readonly '16': "16px";
    readonly '24': "24px";
    readonly '32': "32px";
    readonly '40': "40px";
    readonly '48': "48px";
    readonly '64': "64px";
    readonly '80': "80px";
    readonly '96': "96px";
    readonly '160': "160px";
};
export declare type TSpacing = keyof typeof spacing;
