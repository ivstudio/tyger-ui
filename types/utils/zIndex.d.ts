export declare const zIndex: {
    readonly mobileStepper: 1000;
    readonly appBar: 1100;
    readonly drawer: 1200;
    readonly modal: 1300;
    readonly snackbar: 1400;
    readonly tooltip: 1500;
};
export declare type TZIndex = keyof typeof zIndex;
