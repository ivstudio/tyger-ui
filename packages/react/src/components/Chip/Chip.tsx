'use client';
import styled from 'styled-components';

import { buttonBase, flexCenterInline } from '../../styles';

export type TSize = 'sm' | 'md';

type TChipVariant = 'filled' | 'outlined';

type TChipTags = 'span' | 'div' | 'button';

interface IChip {
    id?: string;
    tag?: TChipTags;
    size?: TSize;
    label: React.ReactNode;
    onClick?: () => void;
    variant?: TChipVariant;
    className?: string;
    disabled?: boolean;
}

const sizes = {
    sm: {
        height: 24,
        padding: `0 8px`,
    },
    md: {
        height: 32,
        padding: `0 12px`,
    },
} as const;

export const ChipRoot = styled.div<{
    size: TSize;
    variant: TChipVariant;
    disabled: boolean;
}>`
    outline: 0;
    line-height: 1;
    white-space: nowrap;
    text-decoration: none;
    vertical-align: middle;
    box-sizing: border-box;
    font-size: 0.8125rem;
    border-radius: 16px;
    ${flexCenterInline};
    ${({ size }) => sizes[size]};

    ${({ theme, variant, disabled }) => {
        if (!theme || !variant) return null;
        const { chip } = theme;
        switch (variant) {
            case 'outlined':
                return disabled
                    ? {
                          color: chip.outlined.disabled.color,
                          background: chip.outlined.disabled.backgroundColor,
                          border: `1px solid ${chip.outlined.disabled.border}`,
                      }
                    : {
                          color: chip.outlined.color,
                          background: chip.outlined.backgroundColor,
                          border: `1px solid ${chip.outlined.border}`,
                      };
            case 'filled':
            default:
                return disabled
                    ? {
                          color: chip.filled.disabled.color,
                          background: chip.filled.disabled.backgroundColor,
                      }
                    : {
                          color: chip.filled.color,
                          background: chip.filled.backgroundColor,
                      };
        }
    }}
`;

export const ChipButtonRoot = styled(ChipRoot)`
    ${buttonBase};
    text-transform: none;

    ${({ theme }) => {
        if (!theme) return null;
        const { chip } = theme;

        return {
            color: chip.button.color,
            background: chip.button.backgroundColor,
            '&:hover': {
                background: chip.button.hover.backgroundColor,
            },
            '&:disabled': {
                color: chip.button.disabled.color,
                background: chip.button.disabled.backgroundColor,
            },
        };
    }}
`;

const Chip = ({
    id,
    label,
    size = 'md',
    tag = 'span',
    onClick,
    variant = 'filled',
    className,
    disabled = false,
}: IChip) => {
    if (tag === 'button') {
        return (
            <ChipButtonRoot
                id={id}
                as="button"
                type="button"
                size={size}
                onClick={onClick}
                variant={variant}
                className={className}
                disabled={disabled}
            >
                {label}
            </ChipButtonRoot>
        );
    }

    return (
        <ChipRoot
            id={id}
            as={tag}
            size={size}
            variant={variant}
            className={className}
            disabled={disabled}
        >
            {label}
        </ChipRoot>
    );
};

export default Chip;
