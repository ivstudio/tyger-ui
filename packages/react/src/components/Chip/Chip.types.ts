export type ChipSizes = 'sm' | 'md';

export type ChipVariant = 'filled' | 'outlined';

type ChipTags = 'span' | 'div' | 'button';

export interface ChipProps {
    id?: string;
    tag?: ChipTags;
    size?: ChipSizes;
    label: React.ReactNode;
    onClick?: () => void;
    variant?: ChipVariant;
    className?: string;
    disabled?: boolean;
}
