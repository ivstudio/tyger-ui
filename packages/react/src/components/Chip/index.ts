'use client';
export { default } from './Chip';
export * from './Chip';
