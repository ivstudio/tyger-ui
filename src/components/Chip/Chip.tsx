import styled from 'styled-components';

import { buttonBase, flexCenterInline } from '../../styles';

export type TSize = 'sm' | 'md';

type TChipVariant = 'filled' | 'outlined';

type TChipTags = 'span' | 'div' | 'button';

interface IChip {
    id?: string;
    tag?: TChipTags;
    size?: TSize;
    label: React.ReactNode;
    onClick?: () => void;
    variant?: TChipVariant;
    className?: string;
}

const sizes = {
    sm: {
        height: 24,
        borderRadius: 16,
        padding: `0 8px`,
        fontSize: 12,
    },
    md: {
        height: 32,
        borderRadius: 16,
        padding: `0 12px`,
        fontSize: 14,
    },
} as const;

export const ChipRoot = styled.div<{ size: TSize; variant: TChipVariant }>`
    outline: 0;
    line-height: 1;
    white-space: nowrap;
    text-decoration: none;
    vertical-align: middle;
    box-sizing: border-box;
    ${flexCenterInline};
    ${({ size }) => sizes[size]};
    border: ${({ theme: { palette }, variant }) => {
        return variant === 'outlined' && `1px solid ${palette.border}`;
    }};
    background: ${({ theme: { palette }, variant }) => {
        return variant === 'filled' && palette.ui.primary;
    }};
    color: ${({ theme: { palette } }) => palette.text.primary};
`;

export const ChipButton = styled(ChipRoot)`
    ${buttonBase};
    text-transform: none;
    &:hover {
        background-color: ${({ theme: { bg } }) => bg.input};
    }
`;

const Chip = ({
    id,
    label,
    size = 'md',
    tag = 'span',
    onClick,
    variant = 'filled',
    className,
}: IChip) => {
    if (tag === 'button') {
        return (
            <ChipButton
                id={id}
                as="button"
                type="button"
                size={size}
                onClick={onClick}
                variant={variant}
                className={className}
            >
                {label}
            </ChipButton>
        );
    }

    return (
        <ChipRoot
            id={id}
            as={tag}
            size={size}
            variant={variant}
            className={className}
        >
            {label}
        </ChipRoot>
    );
};

export default Chip;
