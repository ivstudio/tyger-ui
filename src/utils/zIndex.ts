export const zindex = {
    mobileStepper: 1000,
    appBar: 1100,
    drawer: 1200,
    modal: 1300,
    snackbar: 1400,
    tooltip: 1500,
};
export type TZIndex = keyof typeof zindex;
