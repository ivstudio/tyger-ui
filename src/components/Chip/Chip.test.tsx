import React from 'react';
import Chip from './Chip';
import { render } from '../../../test/utils/mockProvider';

describe('Chip', () => {
    it('renders Chip correctly', () => {
        const { container } = render(<Chip size="md" label="Chip filled" />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
