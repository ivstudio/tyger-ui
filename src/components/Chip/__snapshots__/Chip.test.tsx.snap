// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip renders Chip 1`] = `
<span
  class="sc-bczRLJ gLiHLa"
>
  Chip filled
</span>
`;
